src/DemoData.hs

```diff
@@ -172,20 +162,27 @@ genCommentVote comments_in_context students = do
     let action = addWithUser . AddCommentVote $ comment ^. _Key
     action student <$> arb
 
-updateAvatar :: User -> URL -> forall m . ActionM m => m ()
-updateAvatar user url = update $ SetUserAvatar (user ^. _Id) url
+updateAvatar :: User -> FilePath -> forall m . ActionM m => m ()
+updateAvatar user file = readImageFile file >>= \case
+    Right pic -> saveAvatar (user ^. _Id) pic
+    Left err -> fail err
```

Comment:
Sorry, missed the demo data part. In this case I am ok just having the `fail` combinator here.